suite.lisp

```diff
@@ -0,0 +1,77 @@
+(in-package #:org.shirakumo.trivial-benchmark)
+
+(defvar *benchmark-packages* (make-hash-table :test 'eq)
+  "A table mapping package objects (which are benchmark packages) to a list of fbound symbols.")
+
+(defun find-benchmark-package (package-designator)
+  "Find the benchmark package designated by PACKAGE-DESIGNATOR and return it, otherwise return NIL if it doesn't exist."
+  (let ((package-object (find-package package-designator)))
+    (if (or (null package-object)
+            (null (nth-value 1 (gethash package-object *benchmark-packages*))))
+        nil
+        package-object)))
+
+(defmacro define-benchmark-package (name &body package-options)
+  "Define a package as if by DEFPACKAGE, except that both the COMMON-LISP and TRIVIAL-BENCHMARK packages are used, and that this package is designated as one containing benchmarks."
+  `(progn
+     (defpackage ,name
+       (:use #:trivial-benchmark #:cl)
+       ,@package-options)
+     (setf (gethash (find-package ',name) *benchmark-packages*) nil)))
+
+(defvar *current-suite-report*)         ; Table mapping fn name -> metric alist
+(defvar *current-benchmark*)            ; Name of current benchmark
+(defvar *current-timer*)                ; Timer used in the current benchmark.
+
+(defmacro measure-forms (&body forms)
+  "Benchmark the execution of the forms FORMS within the context of a DEFBENCH macro. Acts like PROGN."
+  `(with-sampling (*current-timer*)
+     ,@forms))
+
+(defun metrics-alist (timer &optional (computations *default-computations*))
+  (loop :for metric :in (metrics timer)
+        :collect (cons (type-of metric)
+                       (mapcan #'list computations (compute computations metric)))))
+
+(defmacro defbenchmark (name args &body body)
```

Comment:
Definition macros have to start with `define-`, no exceptions. So: `define-benchmark`.
If it mirrors defun then it should return the function name.